replications/service.go

```diff
@@ -316,11 +326,42 @@ func (s service) WritePoints(ctx context.Context, orgID platform.ID, bucketID pl
 	//    We gzip the LP to take up less room on disk. On the other end of the queue, we can send the gzip data
 	//    directly to the remote API without needing to decompress it.
 	egroup.Go(func() error {
+		// Set up an initial batch
+		batches = append(batches, &batch{
+			data:      &bytes.Buffer{},
+			numPoints: 0,
+		})
+
+		currentBatchSize := 0
+		currentBatchIndex := 0
+		gzw := gzip.NewWriter(batches[0].data)
+
+		// Iterate through points and compress in batches
 		for _, p := range points {
+			// If current point will cause this batch to exceed max size, start a new batch for it first
+			if currentBatchSize+p.StringSize() > s.maxRemoteWriteBatchSize {
+				currentBatchIndex += 1
+
+				batches = append(batches, &batch{
+					data:      &bytes.Buffer{},
+					numPoints: 0,
+				})
+
+				if err := gzw.Close(); err != nil {
+					return err
+				}
+				currentBatchSize = 0
+				gzw = gzip.NewWriter(batches[currentBatchIndex].data)
```

Comment:
Instead of keeping track of `currentBatchIndex`, would it be simplier to use `len(batches)-1` for the index? Seems like this is only ever accessing the last element in the slice.